tensorrt_llm/configure/cli.py

```diff
@@ -0,0 +1,92 @@
+from pathlib import Path
+from typing import Optional, Union
+
+import yaml
+from pydantic import AliasChoices, BaseModel, Field, model_validator
+from pydantic_settings import BaseSettings, CliSubCommand, SettingsConfigDict, get_subcommand
+
+from tensorrt_llm.configure.profile import InferenceMaxProfile, ThroughputLatencySLAProfile
+from tensorrt_llm.logger import logger
+
+
+class CommonOptions(BaseModel):
+    """Common options for all subcommands of the trtllm-configure CLI tool."""
+
+    output: Optional[Path] = Field(
+        default=None,
+        description="YAML file path where the optimized config will be written.",
+        validation_alias=AliasChoices("output", "o"),
+    )
+
+    @model_validator(mode="after")
+    def validate_output(self) -> "CommonOptions":
+        """Verify that output file is a valid YAML file path and does not already exist."""
+        if self.output is not None:
+            if self.output.suffix != ".yaml":
+                raise ValueError(f"Output file must be a YAML file. Got '{self.output}'.")
+            if self.output.exists():
+                raise ValueError(f"Output file '{self.output}' already exists.")
```

Comment:
if there's an easy way to parametrize and expose a `--force-overwrite` or something, it would help in avoiding a potential user gripe of having to explicitly delete the file everytime - but P1 feel free to ignore